### PR TITLE
Add convenience builder for git commands (smaller)

### PR DESCRIPTION
**Pull request by jesseduffield**
Adds a convenience builder for git commands. Applies it in a couple of places that benefit from it.

I've got a larger [PR](https://github.com/jesseduffield/lazygit/pull/2645) that does a more sweeping change but I want some reviews on that before I go ahead with it.

I'm about to make changes to the code that this PR touches so I want this in master to avoid a rebase dance.

- **Please check if the PR fulfills these requirements**

* [ ] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [ ] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs (specifically `docs/Config.md`) have been updated if necessary
* [ ] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/commands/git_commands/branch_test.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 
 	"github.com/go-errors/errors"
 	"github.com/jesseduffield/lazygit/pkg/commands/oscommands"
+	"github.com/jesseduffield/lazygit/pkg/config"
```
Comment (jesseduffield) [inline pkg/commands/git_commands/branch_test.go:8]: tests aren't directly related to other changes in this PR but worth including

**Comment by jesseduffield** [comment]
Changed my mind: I'm just gonna stick with the original PR until we've settled on the right pattern

File: pkg/commands/git_commands/branch_test.go (hunk 1 above)
Comment (stefanhaller) [reply pkg/commands/git_commands/branch_test.go:8]: In that case my preference would be to have them in a separate "increase test coverage" commit (before the other one).

File: pkg/commands/git_commands/commit_loader.go [hunk 2]
```diff
@@ -413,9 +422,10 @@ func ignoringWarnings(commandOutput string) string {
 func (self *CommitLoader) getFirstPushedCommit(refName string) (string, error) {
 	output, err := self.cmd.
 		New(
-			fmt.Sprintf("git merge-base %s %s@{u}",
-				self.cmd.Quote(refName),
-				self.cmd.Quote(strings.TrimPrefix(refName, "refs/heads/"))),
+			NewGitCmd("merge-base").
+				Arg(self.cmd.Quote(refName)).
```
Comment (stefanhaller) [inline pkg/commands/git_commands/commit_loader.go:426]: Wonder if it's worth having `QuotedArg` for this?

File: pkg/commands/git_commands/git_command_builder.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func NewGitCmd(command string) *GitCommandBuilder {
+	return &GitCommandBuilder{command: command}
+}
+
+func (self *GitCommandBuilder) Arg(flag string) *GitCommandBuilder {
+	if flag == "" {
+		return self
+	}
```
Comment (stefanhaller) [inline pkg/commands/git_commands/git_command_builder.go:18]: Is this really needed?

**Pull request closed without merging**